.github/workflows/check-lint-udeps-test.yaml

```
@@ -0,0 +1,133 @@
+# Basic flow where we do all generic tests:
+#   - `cargo check`: Check that our crate compiles
+#   - `cargo fmt --all -- --check`: Checks that our crate follows basic Rust formatting
+#   - `cargo workspaces exec cargo +nightly udeps`: Makes sure that the branch does not have any unused dependencies
+#   - `cargo test`: Makes sure tests are passing
+
+# Basic checks should run on PR on `develop` or `main`
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
+
+
+name: Check and Lint
```

I would suggest the following:

```suggestion
name: Integrate
```